crates/shared/src/zeroex_api.rs

```diff
@@ -196,23 +190,46 @@ pub struct Order {
     /// anyone can fill the order.
     pub taker: H160,
     /// The amount of `taker_token` being sold by the taker.
-    pub taker_amount: U256,
+    #[serde(with = "serde_with::rust::display_fromstr")]
+    pub taker_amount: u128,
     /// The address of the ERC20 token the taker is selling to the maker.
     pub taker_token: H160,
     /// Amount of takerToken paid by the taker to the feeRecipient.
-    pub taker_token_fee_amount: U256,
+    #[serde(with = "serde_with::rust::display_fromstr")]
+    pub taker_token_fee_amount: u128,
     /// Address of the contract where the transaction should be sent, usually this is
     /// the 0x exchange proxy contract.
     pub verifying_contract: H160,
 }
 
 #[derive(Debug, Default, Clone, Deserialize, PartialEq)]
-#[serde(rename_all = "camelCase")]
 pub struct OrderRecord {
+    #[serde(rename = "metaData")]
     pub metadata: OrderMetadata,
     pub order: Order,
 }
 
+impl OrderRecord {
+    /// Scales the `maker_amount` according to how much of the partially fillable
+    /// amount was already used.
+    pub fn remaining_maker_amount(&self) -> Result<u128> {
+        let to_big_int = |num: u128| BigInt::from_u128(num).unwrap();
```

Comment:
Might actually make sense to use `U256` checked math here instead of `BigInt` since that is what the SC on-chain does. (Also saves some heap allocations 🎉).

That being said, I don't think anything here can overflow a `U256` so I think its OK as is as well.

Reply:
Good idea. This also made the good nicer to look at because `BigInt` is somewhat awkward to convert to and from.